nlprule/src/compile/impls.rs

```diff
@@ -145,18 +145,25 @@ impl Tagger {
             .map(|(i, x)| (x.to_string(), PosIdInt::from_value_unchecked(i as u16)))
             .collect();
 
-        let mut tags: WordIdMap<Vec<(WordIdInt, PosIdInt)>> = WordIdMap::new(word_store.len());
+        let mut tags: Vec<Option<Vec<(WordIdInt, PosIdInt)>>> = vec![None; word_store.len()];
 
         for (word, inflection, tag) in lines.iter() {
             let word_id = word_store.get_by_left(word).unwrap();
             let lemma_id = word_store.get_by_left(inflection).unwrap();
             let pos_id = tag_store.get_by_left(tag).unwrap();
 
-            tags.get_mut_or_default(*word_id).push((*lemma_id, *pos_id));
+            match &mut tags[word_id.value() as usize] {
```

Comment:
`tags.get_mut()`

Reply:
Thanks for taking a look.

- Unfortunately with `tags.get_mut()` we would need `tags
                .get_mut(word_id.value() as usize)
                .map(|x| x.as_mut())
                .flatten()` since we'd get an `Option<&mut Option<..>>`.
- I don't like magic numbers and `with_capacity` where it is not needed: the compile module is the least performance sensitive code since it is only used to build the binaries and there is no obvious upper limit on the number of tags.

Reply:
A need you are entirely correct, I misread inner and outer vec manipulation. Discard all comments. Sorry.

Reply:
No problem. I'll look over this once more tomorrow and merge it then.